broker-util/oo-admin-repair

```diff
@@ -403,6 +418,37 @@ def find_unresponsive_servers(available_servers, auto_confirm)
   unresponsive_servers
 end
 
+def find_orphaned_envs(namespace=nil)
+  orphaned_envs = []
+  domains = []
+  if namespace
+    begin
+      domains = [Domain.find_by(:namespace => namespace)]
+    rescue Mongoid::Errors::DocumentNotFound
+      print_message "Unable to find domain. The domain #{namespace} does not exist. Skipping..."
+    rescue Exception => e
+      print_message "Unable to find domain #{namespace} with error: #{e.message}."
+    end
+    # Find all component_ids for every application in this domain
+    component_ids = domains[0].applications.map {|a| a.component_instances.map {|ci| ci._id}}.flatten
+  else
+    domains = Domain.all
+    # find all component_ids for every application
+    component_ids = Application.all.map {|a| a.component_instances.map {|ci| ci._id}}.flatten
```

Comment:
I worry a little bit about an array that will contain all the applications in an environment as big as OpenShift Online.  It's seems OK to run this to repair a given Domain but I worry about the admin that runs this thinking it will fix up everything only to see if OOM or impact the performance on the machine running the tool.

There's probably a much more efficient query you could do in Mongo to get just the data you want.  A simplistic approach would be `Application.only('component_instances._id').all`.  The weird thing is that you will get back a Application objects where just about everything is nil except for the component_instances _id.  That will probably save quite a bit of system resources.  I would play around with it a bit to see if you could find something even better.